code/obj/machinery/computer/cloning.dm

```diff
@@ -887,6 +652,173 @@
 				set_lock(0)
 				active_process = PROCESS_IDLE
 
+/obj/machinery/computer/cloning/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("delete")
+			var/selected_record =	find_record(params["ckey"])
+			if(selected_record)
+				logTheThing("combat", usr, null, "deletes the cloning record [selected_record["fields"]["name"]] for player [selected_record["fields"]["ckey"]] at [log_loc(src)].")
+				src.records.Remove(selected_record)
+				qdel(selected_record)
+				selected_record = null
+				src.temp = "Record deleted."
+				. = TRUE
+		if("scan")
+			if(!isnull(src.scanner))
+				src.scan_mob(src.scanner.occupant)
+				. = TRUE
+		if("clone")
+			var/ckey = params["ckey"]
+			if(ckey)
+				clone_record(find_record(ckey))
+				. = TRUE
+		if("toggleGeneticAnalysis")
+			if(pod1 && !pod1.operating)
+				pod1.gen_analysis = !pod1.gen_analysis
+				. = TRUE
+			else
+				src.temp = "Cannot toggle any modules while cloner is active."
+				. = TRUE
+		if("saveToDisk")
+			var/ckey = params["ckey"]
+			var/selected_record = find_record(ckey)
+			if ((isnull(src.diskette)) || (src.diskette.read_only) || (isnull(selected_record)))
+				src.temp = "Save error."
+				return
+
+			for (var/datum/computer/file/clone/R in src.diskette.root.contents)
+				if (R.fields["ckey"] == selected_record["fields"]["ckey"])
+					src.temp = "Record already exists on disk."
+					return
```

Comment:
```suggestion
					return TRUE
```
even though this is returning early it's still changing the UI so we should probably still return TRUE